### PR TITLE
Modernize step-29.

### PR DESCRIPTION
**Pull request by drwells**
1. Inline some function definitions
2. Use ranged-for loops
3. Minor typography fixes (e.g., delete trailing whitespace)
4. Update console output

### CODE REVIEW

File: examples/step-29/doc/intro.dox [hunk 1]
```diff
@@ -96,26 +96,26 @@ boundary condition will yield partial reflections, i.e. only parts of the wave
 will pass through the boundary as if it wasn't here whereas the remaining
 fraction of the wave will be reflected back into the domain.
 
-If we are willing to accept this as a sufficient approximation to an absorbing boundary we finally arrive at the following problem for $u$: 
+If we are willing to accept this as a sufficient approximation to an absorbing boundary we finally arrive at the following problem for $u$:
```
Comment (kronbichler) [inline examples/step-29/doc/intro.dox]: Want to break the line around column 80?

File: examples/step-29/doc/results.dox [hunk 2]
```diff
@@ -43,33 +43,39 @@ parameters here and therefore stick with their default values.
...
 DEAL::  Number of active cells:  25600
-DEAL::Setting up system... done (1.10807s)
+DEAL::Setting up system... done (0.614761s)
 DEAL::  Number of degrees of freedom: 51842
-DEAL::Assembling system matrix... done (6.50841s)
-DEAL::Solving linear system... done (2.89218s)
-DEAL::Generating output... done (4.52428s)
+DEAL::Assembling system matrix... done (2.82536s)
+DEAL::Solving linear system... done (2.27971s)
+DEAL::Generating output... done (1.84418s)
```
Comment (kronbichler) [inline examples/step-29/doc/results.dox]: It's pretty interesting to see how different components (setup, assembly, solver) have evolved differently in terms of their time (and your system versus the one originally used).

File: examples/step-29/doc/intro.dox (hunk 1 above)
Comment (drwells) [reply examples/step-29/doc/intro.dox]: I wrapped this and all the other long lines (and long equations) .in this file.

**Pull request merged**